hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -338,9 +338,9 @@ public void memberAdded(MemberImpl member) {
         logger.fine("Adding " + member);
         lock.lock();
         try {
-            if (!member.localMember()) {
+//            if (!member.localMember()) {
```

Comment:
I think you're right about cluster state. Lite member promotion should not be allowed when migrations are not allowed by cluster state. I'll send a new PR to fix that.